### DISCUSSION TITLE
Should is_verified be True for OAuth users

### DISCUSSION THREAD

**Discussion by KentonParton** [fastapi-users/fastapi-users#1148] [Q&A]
When a user signs in for the first time via an OAuth provider, the user is created in your DB and `is_verified` is false. Would we not expect the user to be verified since verification is typically performed by the OAuth provider? Thanks!

**Comment by frankie567** [comment]
Well, in theory, I would say yes. However, I think that some people would like to still ask for verification though. Maybe it could be a setting? What do you think?

**Reply by KentonParton** [reply]
I'd agree, I think some people would still like to ask for verification. Could we achieve this by modifying our user models? We would have to check if length of `oauth_accounts` > 1 then set `is_verified` to true else false.

A setting would be nice too but if it's easy enough to update the user models and it's shown in the docs that sounds great too!

**Reply by frankie567** [reply]
That could work maybe, but IMO, it would give a bit too much responsibility to the model. The code creating the user after OAuth sign-up is here:

https://github.com/fastapi-users/fastapi-users/blob/0e975c9b95c1b4387fcb30eb33f6d946efc086ec/fastapi_users/manager.py#L205-L213

I think the solution is to add a new keyword parameter to `oauth_callback` (like `associate_by_email`) to tell if we want to set the verify flag to `True` or `False` upon registration (with a default to `False` to keep the current behavior).  Then, we would just have to set it in `is_verified` to this value in `user_dict`.

**Reply by KentonParton** [reply]
I like the sound of that 👍🏼

**Comment by frankie567** [comment]
This is now available in [10.4](https://github.com/fastapi-users/fastapi-users/releases/tag/v10.4.0) 😄

**Reply by KentonParton** [reply]
Amazing, thanks François!
